[AIDB] NIM content for Using and reorg (#6605)
EnterpriseDB/docs: advocacy_docs/edb-postgres-ai/ai-accelerator/models/using-with/using-nvidia-nim/using-nim-in-your-environment.mdx

---
title: Using Nvidia NIM model in your environment
navTitle: In your environment
description: Learn how to use Nvidia NIM models in your environment, locally or in your private cloud.
---

> To use a Nvidia NIM that is hosted in your own environemtn, you first need a instance of the model. For this tutorial, we will show how to configure an AWS hosted instance with the Nvidia NIM model. In this tutorial, we will use the Nvidia NIM model llama3-8b-instruct.

Reviewer: ```suggestion
To use a Nvidia NIM that is hosted in your own environment, you first need a instance of the model. For this tutorial, we will show how to configure an AWS hosted instance with the Nvidia NIM model. In this tutorial, we will use the Nvidia NIM model llama3-8b-instruct.
```

## Prerequisites

> * A system capable of running Nvidia CUDA Toolkit. We recommend, for this tutorial, using an **EC2 g5.8xlarge instance** with **1024 GB of gp3 storage** running **Ubuntu 24.04 LTS**.

Reviewer: ```suggestion
* A system capable of running Nvidia CUDA Toolkit. We suggest, for this tutorial, using an **EC2 g5.8xlarge instance** with **1024 GB of gp3 storage** running **Ubuntu 24.04 LTS**, although smaller instance sizes may also work.
```

```shell
# Choose a container name
export CONTAINER_NAME=Llama3-8B-Instruct
NGC_API_KEY="<NGC API KEY>"
# Define the repository and image
Repository=nim/meta/llama3-8b-instruct
export IMG_NAME="nvcr.io/${Repository}:latest"
# Set local cache path
export LOCAL_NIM_CACHE=~/.cache/nim
mkdir -p "$LOCAL_NIM_CACHE"
# Start the LLM NIM container
docker run -it --rm --name=$CONTAINER_NAME \
--runtime=nvidia \
--gpus all \
--shm-size=16GB \
-e NGC_API_KEY=$NGC_API_KEY \
-v "$LOCAL_NIM_CACHE:/opt/nim/.cache" \
-u $(id -u) \
-p 8000:8000 \
$IMG_NAME 
```

### 6. Test the Deployment

Run the following command to verify the model is working:

```shell
curl -X POST "http://0.0.0.0:8000/v1/chat/completions" \
    -H "Accept: application/json" \
    -H "Content-Type: application/json" \
    -d '{
        "model": "meta/llama3-8b-instruct",
        "messages": [
        {
            "role": "user",
            "content": "Tell me a story"
        }
        ]
    }'
```

> Once you have verified the model is working, you can use it in with EDB Postgres AI Accelerator.

Reviewer: ```suggestion
Once you have verified the model is working, you can use it with EDB Postgres AI Accelerator.
```

Make a note of the public IP address of the EC2 instance, as you will need it to connect to the model from EDB Postgres AI Accelerator. This will be referred to as `<NIM_HOST>` in the following steps.

## Integrating the Model with AI Accelerator

### 1. Enable AI Accelerator in EDB Postgres AI

> With an EDB Postgres Advanced Server instance running, connect to the database and enable the AI Accelerator extension:

Reviewer: ```suggestion
With an EDB Postgres Advanced Server instance running, connect to the database and enable the AI Accelerator extension by running the following SQL commands:
```
